src/components/auth/VerifyAccount.jsx

```diff
@@ -0,0 +1,36 @@
+import { Component } from "react";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import { withRouter } from "react-router-dom";
+import verifyUserEmail from "<actions>/auth/verifyUser";
+
+class VerifyAccount extends Component {
+  componentDidMount() {
+    const {
+      location: { search }
+    } = this.props;
+    this.props
+      .verifyUserEmail(search)
+      .then(() => {
+        this.props.history.push("/profile");
+      })
+      .catch(() => {
+        this.props.history.push("/");
+      });
+  }
+
+  render() {
+    return null;
```

Comment:
What is this?

Reply:
This goes against React principles. This probably is better in the route as `renderProps`